rmf_traffic_editor/gui/delete_dialog.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "delete_dialog.h"
+#include <algorithm>
+#include <QtWidgets>
+using std::vector;
+using std::string;
+
+
+DeleteDialog::DeleteDialog(
+  QWidget* parent)
+: QDialog(parent)
+{
+  setWindowTitle("Warning");
+  QVBoxLayout* warning_message_vbox_layout = new QVBoxLayout;
+  warning_message_vbox_layout->addWidget(new QLabel(
+      "This item might be linked to other entities and/or on other levels.\n "
+      "Are you sure you want to delete this item?"));
```

Comment:
It would be nice to make the messaging here a bit more specific so the user really knows what is going to happen. I tried the feature and while it works it ended up deleting _a lot_ of entities when I deleted a vertex, this is especially dangerous since `undo` for vertex deletion doesn't really work and users would probably have to reload their project if they delete the wrong vertex.
Since there are different code paths being executed to check usage (from what I see at least lifts, edges and polygons all have different paths) it should be possible to populate a list of entities (or at the very least type of entities with number) that could be affected, so TLDR best scenario something like:

```
This vertex is being used by 10 entities:
* Level [L2]:
  * 1 graph lane(s) for graph index [0]
  * 3 graph lane(s) for graph index [1]
  * 2 walls
  * 1 lift with lift name [lift_1]
  * 1 polygon
*  Level [L2]
  * [...] 
```

Slightly less informative but still better than current

```
This vertex is being used by 10 entities:
* Level [L1]:
  * 5 edges
  * 1 lift
  * 1 polygon
*  Level [L2]
  * [...] 
```

Least informative

```
This vertex is being used by 10 entities:
```

Reply:
I have made the neccessary changes to achieve something similar to the best scenario.

But it is tailored to two different scenarios.

For lift cabin, it will inform the user that the vertex is used across different levels (may or may not have lanes connected to them) and show the different levels that the lift has access to.

For normal edges and polygons, it is not possible to have inter-level references. So when a user wants to delete a used vertex, the warning will show all the different types of edges and polygons that are currently using said vertex.

Reply:
![Screenshot 2024-12-18 at 10 29 00 AM](https://github.com/user-attachments/assets/257ca34c-1ade-4498-8396-e87ec29a0536)
![Screenshot 2024-12-18 at 10 32 16 AM](https://github.com/user-attachments/assets/03addafb-4eed-48cc-8d50-36ce6211a85c)